pipelines/utils/utils.py

```
@@ -83,3 +87,40 @@ def to_partitions(
+                    "\n",
+                    exc,
+                )
+                raise
+    return dataframe
+
+
+def download_and_unzip_file(url: str, path: str) -> None:
+    """Baixa um arquivo ZIP da URL e extrai no caminho indicado."""
+    log(f"Baixando {url} → {path}")
+    try:
+        r = urlopen(url)
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)